apps/researcher/src/app/[locale]/objects/[id]/object-lists-actions.ts

```diff
@@ -0,0 +1,35 @@
+'use server';
+
+import {getCommunityBySlug} from '@/lib/community';
+import {objectList} from '@colonial-collections/database';
+import {InsertObjectItem} from '@colonial-collections/database';
+import {revalidatePath} from 'next/cache';
+
+export async function getCommunityLists(communityId: string, objectId: string) {
+  return objectList.getByCommunityId(communityId, {objectIri: objectId});
+}
+
+interface AddObjectToListProps {
+  objectItem: InsertObjectItem;
```

Comment:
Naming-wise this is still a bit confusing to me - I would expect an 'object item' to be of type 'ObjectItem' instead of 'InsertObjectItem'. Do you see a way to change this?

Reply:
I can rename it on line 5. However, I would like to keep the types `InsertObjectItem` and `SelectObjectItem`. Because for the insert `id` and `createdAt` are optional. For the `SelectObjectItem` it is not.

Reply:
See commit: https://github.com/colonial-heritage/colonial-collections/pull/272/commits/7562b4565938830ddef71d51fa277023f195b193

Is this ok?

Reply:
An alias, that's a creative solution :) But I don't think it solves the real issue, though: the naming and typing of things.

Currently the types originate from the `database` package and are used in the frontend. This leads to tight-coupling. For example: the names `InsertObjectItem` and `SelectObjectItem` are 'database terminology'. But I think the frontend should be able to use types that are storage-agnostic - hence the generic entity name `ObjectItem`. (The fact that the database package uses it to insert or to select an object item is an implementation detail then.)

Yes, I fully agree that you want to keep different types for insertion and selection. But from an API perspective I would use more generic names that identify the nature/state of the object item. For example: `ObjectItem` for a full object item with all properties and `ObjectItemBeingCreated` for an object item that doesn't have all props yet.

Having said all this, feel free to not implement this, or not now (the ~~show~~ PR must go on).